src/components/Tables/Table.tsx

```diff
@@ -0,0 +1,99 @@
+"use client";
+
+import React, { useState, useMemo } from "react";
+import DataTable, { TableColumn } from "react-data-table-component";
+
+interface Props {
+    data: any[];
+    headers: { [key: string]: string | null };
+}
+
+interface FilterComponentProps {
+    filterText: { [key: string]: string };
+    onFilter: (e: React.ChangeEvent<HTMLInputElement>, filterField: string) => void;
+    onClear: () => void;
+    headers: { [key: string]: string | null };
+}
+
+const FilterComponent: React.FC<FilterComponentProps> = ({
+    filterText,
+    onFilter,
+    onClear,
+    headers,
+}) => {
+    return (
+        <>
+            {Object.entries(headers).map(([key, value]) => {
+                return (
+                    <input
+                        key={key}
+                        type="text"
+                        value={filterText[key]}
+                        placeholder={`Filter by ${value}`}
+                        onChange={(e) => onFilter(e, key)}
+                    />
+                );
+            })}
+            <button onClick={onClear}>Clear</button>
+        </>
+    );
+};
```

Comment:
Probably best to have this in a separate "FilterComponent" file instead of grouping the implementation with the Table component. You can also then rewrite "FilterComponentProps" as simply "Props" within the new file. Would improve clarity and modularity of code

Reply:
We have separated out the component as recommended.